### ISSUE TITLE
Browser freezes when typing in URL bar

### ISSUE THREAD

**Issue by luixxiul** [brave/browser-laptop#11066]
### Description

https://community.brave.com/t/browser-freezes-when-typing-in-address-bar/6336?u=suguru

> I just installed Brave today (ver. 0.18.23), and everything is working fine except the browser freezes for a second or two every time I type something into the address/search bar. It always unfreezes and catches up to my typing, but it’s a bit annoying.

### Steps to Reproduce

  1.
  2.
  3.

**Actual result:**

**Expected result:**

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Brave Version

**about:brave info:**

> 0.18.23

**Reproducible on current live release:**

### Additional Information

I also experience the issue with my existing browser's profile.

**Comment by luixxiul** [comment]
@zentagonist would you mind triaging in case this issue is widely reproduced?

**Comment by mmdmo2090** [comment]
Hey, I had this same issue. I removed all tab settings under preferences and it cleared it up. I don't know if its a bug specifically on macOS, but I compared it to my windows PC and didn't have any issues with the settings being on.

![image](https://user-images.githubusercontent.com/25968009/31021719-abaf51e8-a504-11e7-837b-22994e0b4d2d.png)

Hope this helps.

**Comment by stormy706** [comment]
Deleting my browser history (imported from FF) solved the issue, however it wasn't really good that I had to do that...

**Comment by zbroinx** [comment]
@stormy706 
Thank you. After delete imported browser history, solved my problem.

**Comment by JulesPalmer** [comment]
My issue with the browser freezing when typing into the URL bar was resolved by switching off "Show history matches" on the Search tab in Preferences.
The history I imported from FF was probably too much. Hope this helps.

**Comment by stormy706** [comment]
Yes exactly, I maybe it doesnt index important history properly, causing these load times (as it sifts through the old records)

**Comment by WillMcK** [comment]
Just like to add that this is still occurring. It happens when you open a new tab and start typing in the address bar. It will stall for about 5-10 seconds before letters show up.

Brave Version: **0.22.721**

Operating System: **MacOS 10.13.4**


Bandaid: @mmdmo2090 suggestion worked for me. Go to Preferences > Search > then turn off all Search Bar Options.

**Comment by strathmeyer** [comment]
I can confirm that this still happens. I imported my browser history from Chrome and saw 10 seconds of spinning beach ball after I typed the first character in a new tab. Even typing the same characters in another new tab would cause a big delay.

Turning off "Show history matches" in Search preferences removed the delay. I was also able to remove the delay by deleting Browsing history. (Oddly, even though I used the "Clear Browsing Data Now…" button and did not have any of the "clear when I close" options selected, my browsing history did not disappear until I restarted Brave.)

MacOS 10.13.5
---------------------
Brave: 0.23.39 
V8: 6.7.288.46 
rev: 3ee14b026f09d7d71c1985b4a9c9a11d78ab8f49 
Muon: 7.1.6 
OS Release: 17.6.0 
Update Channel: Release 
OS Architecture: x64 
OS Platform: macOS 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 67.0.3396.103
